### DISCUSSION TITLE
Apollo Server running but failed to fetch

### DISCUSSION THREAD

**Discussion by arsekil** [keystonejs/keystone#8409] [Questions] [answered]
Hello team,

I have a weird problem here, that when I try querying some data from Keystone through GraphQL playground into Next.js frontend, it always returns `Failed to fetch`. Although, the keystone server is running and the playground is accessible from localhost.

My Next.js dev server is running on `port 8000`.

I have this code inside my ApolloClient.js file:
``` typescript
import { ApolloClient, InMemoryCache } from "@apollo/client";

const client = new ApolloClient({
   uri: "http://localhost:3000/api/graphql",
   cache: new InMemoryCache(),
});
```

`export default client;`

I tried even without the `api/graphql` part, but no luck since. In the playground is working just fine. 

I have read some answers on Stackoverflow and they said i might need to configure cors for this, because i just checked, it's a CORS error I am being given in the Chrome Dev Console.

Did anyone encounter such problem and how did you overcame it?

Thank you.

**Comment by arsekil** [comment, marked as answer]
I found the solution meanwhile.

I had to configure CORS in the keystone.ts file, inside the  ServerConfig type.

It's working nince and dandy.

Cheers.

**Reply by dcousens** [reply]
@andrejmoltok for posterity, did you want to share what your solution was?

**Reply by arsekil** [reply]
Here is the docs about it, which I used.

https://keystonejs.com/docs/config/config#server
